commerce_coordinator/apps/lms/views.py

```diff
@@ -482,16 +483,30 @@ def get(self, request, bundle_key=None):
         user_purchasable_course_keys = request.GET.getlist("course_key")
 
         try:
+            view_start_time = datetime.datetime.now()
             ct_api_client = CTCustomAPIClient()
 
             # Fetch bundle variants with entitlements
+            start_time = datetime.datetime.now()
             ct_program_variants = ct_api_client.get_program_variants(bundle_key)
+            duration = (datetime.datetime.now() - start_time).total_seconds()
+            logger.info(
+                f"[Performance Check - ProgramPriceView] - ct_program_variants took {duration} seconds. "
+                f"Bundle key: {bundle_key}"
+            )
+
             if not ct_program_variants:
                 logger.error(f"[ProgramPriceView] No program variants found for the program: "
                              f"{bundle_key} for user: {username}.")
                 return Response('Program variants not found.', status=HTTP_404_NOT_FOUND)
 
+            start_time = datetime.datetime.now()
             ct_discount_offers = ct_api_client.get_ct_bundle_offers_without_code()
+            duration = (datetime.datetime.now() - start_time).total_seconds()
+            logger.info(
+                f"[Performance Check - ProgramPriceView] - get_ct_bundle_offers_without_code took {duration} seconds. "
+                f"Bundle key: {bundle_key}"
+            )
 
             # Get applied program offer
             program_offer = get_program_offer(ct_discount_offers, bundle_key)
```

Comment:
Don't we need to check its performance?

Reply:
I dont think so because there is no external API involved. Its filtration code only.